src/overridable.js

```diff
@@ -0,0 +1,86 @@
+import React, {useContext} from 'react';
+import PropTypes from 'prop-types';
+
+// create a new context with an empty map of overridden components as default value.
+export const OverridableContext = React.createContext({});
+
+/**
+ * Function to wrap a React component and override default props.
+ * @param Component the component to wrap
+ * @param extraProps the new props that will override or will be added to the wrapped component
+ * @returns the wrapper component
+ */
+export function parametrize(Component, extraProps) {
+  const ParametrizedComponent = props => {
+    // handle deferred prop calculation
+    if (typeof extraProps === 'function') {
+      extraProps = extraProps(props);
+    }
+
+    // Store the original component in an attribute
+    if (Component.originalComponent) {
+      Component = Component.originalComponent;
+    }
+
+    // overrideProps override props if there is a name collision
+    const {children, ...attrProps} = {...props, ...extraProps};
+    return React.createElement(Component, attrProps, children);
+  };
+
+  const name = Component.displayName || Component.name;
+  ParametrizedComponent.displayName = `Parametrized(${name})`;
+  return ParametrizedComponent;
+}
+/**
+ * React component to enable overriding children when rendering.
+ */
+function Overridable({id, children, ...restProps}) {
+  const child = React.Children.only(children); // throws an error if multiple children
+  const childProps = child.props || {};
+  const overriddenCmps = useContext(OverridableContext);
+  const Overridden = overriddenCmps[id];
```

Comment:
this should fix it (also moved up the hook call to keep them at the beginning)

```suggestion
  const overriddenCmps = useContext(OverridableContext);
  const child = children ? React.Children.only(children) : null;
  const childProps = child ? child.props : {};
  const Overridden = overriddenCmps[id];
```